requirements.txt

```diff
@@ -0,0 +1,3 @@
+traitlets
```

Comment:
So we specify `notebook` in `install_requires` in `setup.py`, which pulls in `traitlets` as a transitive dependency, so it shouldn't be necessary here. Since `pytest` isn't required when running the code, it is often listed in a `dev-requirements.txt` instead, and we add installing that to `docs/contributing.md`. I think the `-e .` is often not set up in `dev-requirements.txt` either, and it's explicitly just specified in `docs/contributing.md` - not sure exactly why, but that is the general consensus I've seen.

Reply:
Some packages behave differently when they're installed in editable mode, so it's better not to force an installation method so a developer has the freedom to test the package installed in production mode (particularly important for CI), or to install in editable mode for development and accept there may be differences.

Reply:
I am with you here. good deal